src/client.ts

```diff
@@ -700,9 +701,15 @@ export class StreamChat<StreamChatGenerics extends ExtendableGenerics = DefaultG
 
   /**
    * getAppSettings - retrieves application settings
+   *
+   * @param [options={}] App settings are cached by default. Pass { force: true } to force cache refresh.
    */
-  async getAppSettings() {
-    return await this.get<AppSettingsAPIResponse<StreamChatGenerics>>(this.baseURL + '/app');
+  async getAppSettings(options: { force?: boolean } = {}) {
+    if (options.force || !this._appSettings) {
+      this._appSettings = await this.get<AppSettingsAPIResponse<StreamChatGenerics>>(this.baseURL + '/app');
+    }
+
+    return this._appSettings;
```

Comment:
In case user-level privacy settings are not provided, the client should fall back to app-level settings. We don't want to fetch app settings on every keystroke event, so adding caching here.

In practice this is not a big change, because the React SDK already does caches app settings on component level. (That caching can be removed and after this PR is merged.)

Reply:
Can you check how other JS SDKs handle this? This could be potentially a breaking change.